Formula/infer.rb

```diff
@@ -29,14 +29,13 @@ def install
     ENV["OPAMROOT"] = opamroot
     ENV["OPAMYES"] = "1"
 
-    system "opam", "init", "--no-setup"
-    system "opam", "update"
+    # Some of the libraries installed by ./build-infer.sh do not
+    # support parallel builds. Homebrew sets this variable to -j<ncpu>
+    # by default, which would break the build. Moreover,
+    # ./build-infer.sh has its own parallelization logic.
+    ENV["MAKEFLAGS"] = ""
```

Comment:
I didn't know about `ENV.j1`, that's exactly what I want here, thanks.

The program which doesn't support parallel compilation is ocaml itself, and it's a known issue: https://github.com/ocaml/ocaml/pull/442

The `./build-infer.sh` script attempts to mitigate this by compiling other dependencies and infer itself in parallel, so not all is lost: https://github.com/facebook/infer/blob/b4f6ea18270ad547056ba6933db5ccf83dda9dc1/build-infer.sh#L17